### ISSUE TITLE
TypeScript definitions for types of `value` returned by `getSelectedDataAsync`

### ISSUE THREAD

**Issue by LukeOwlclaw** [OfficeDev/office-js-docs-reference#1048]
When calling `getSelectedDataAsync` the type of the value received by the callback method depends on the specified coercion type ("The value property of the result is the data in the current selection. This is returned in the data structure or format you specified with the coercionType parameter. (See Remarks for more information about data coercion.)").

I want to call:

```
Office.context.document.getSelectedDataAsync(Office.CoercionType.SlideRange, (asyncResult) => {
    if (asyncResult.status === Office.AsyncResultStatus.Succeeded) {
        const slideRange = asyncResult.value as OSF.DDA.SlideRange;
        Logger.Log(slideRange.slides);
    }
    else {
        Logger.Log("Action failed with error: " + asyncResult.error.message);
    }
});
```

``OSF.DDA.SlideRange`` is the type the debugger console gives for ``asyncResult.value``. Where can I find the TypeScript definition for it? What about the definitions of the return value for the other coercion types?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a80b0c0b-0647-036d-9193-4be6c67edf21
* Version Independent ID: db81f510-204a-d07b-f02f-6b0e83f1f592
* Content: [Office.Document interface - Office Add-ins](https://docs.microsoft.com/en-us/javascript/api/office/office.document?view=excel-js-preview#getSelectedDataAsync_coercionType__callback_)
* Content Source: [docs/docs-ref-autogen/office/office/office.document.yml](https://github.com/OfficeDev/office-js-docs-reference/blob/live/docs/docs-ref-autogen/office/office/office.document.yml)
* Product: **non-product-specific**
* Technology: **add-ins**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by ElizabethSamuel-MSFT** [comment]
@LukeOwlclaw Thanks for asking about this.

@Rick-Kirkham Can you take a look?

Thanks.

**Comment by Rick-Kirkham** [comment]
I remember that we considered creating TypeScript definitions for these return values but cancelled the project. I don't remember why, but probably because there wasn't a lot of customer demand for it and it would be more labor-intensive that it appears to be superficially. I'll investigate again. In the meantime, does this article server your needs? 

[Office.CoercionType](https://docs.microsoft.com/en-us/javascript/api/office/office.coerciontype)

**Comment by LukeOwlclaw** [comment]
@Rick-Kirkham Thanks, for your answer.
I just wanted to make sure that I did not miss the definition. For me the superficial version is fine. From ``{"slides":[{"id":257,"title":"Slide 2","index":2},{"id":256,"title":"Slide 1","index":1}]}`` I created the interface myself. Looks like this:

```
interface ISlide {
    id: number;
    title: string;
    index: number;
}

interface ISlideRange {
    slides: ISlide[];
}
```

Unless you want to extend the docs, I guess we can close this issue.
